### PR TITLE
swev-id: django__django-12406 Fix RadioSelect empty label

### LINKED ISSUE #192: ModelForm RadioSelect for ForeignKey should not show blank when blank=False
## User Request

ModelForm RadioSelect widget for foreign keys should not present a blank option if blank=False on the model.

Unlike the select widget, where a blank option is idiomatic even for required fields, radioselect has an inherent unfilled state that makes the "-------" option look suspiciously like a valid choice.

```python
class TestRun(models.Model):
    data_file = models.ForeignKey(BatchData, on_delete=models.SET_NULL, null=True, blank=False)

class TestRunForm(ModelForm):
    class Meta:
        model = TestRun
        fields = ['data_file']
        widgets = {'data_file': RadioSelect()}
```

Currently renders:

```
<ul id="id_data_file">
 <li><label for="id_data_file_0">
    <input checked="checked" id="id_data_file_0" name="data_file" type="radio" value=""> ---------
 </label></li>
 <li><label for="id_data_file_1">
    <input id="id_data_file_1" name="data_file" type="radio" value="1"> First Data File
 </label></li>
</ul>
```

Instead, there should be no checked option for RadioSelect's `<input>` tags when rendering a new form from a model if blank is not a valid selection.

Reference PR: https://github.com/django/django/pull/11199

## Research Specification (Emerson Gray)

- Current behavior: ModelChoiceField adds an empty choice ("---------") via `empty_label` when `required=True` and initial is `None`. RadioSelect then renders an empty radio with value `''` and marks it checked on fresh forms.
- Change: For ModelChoiceField rendered with `RadioSelect`, when the field is required (`blank=False`), do not include an empty choice; radios support an unselected initial state.
- Code edits:
  - django/forms/models.py: import `RadioSelect`.
  - In `ModelChoiceField.__init__`, set `empty_label = None` when `required` and `isinstance(widget, RadioSelect)`.
- Acceptance criteria:
  - Required RadioSelect (ForeignKey `blank=False`): no radio with `value=""` and no radio checked on new/unbound form.
  - Bound with empty submission (`''`): invalid form; still no empty radio and none checked.
  - With initial valid value: corresponding radio is checked; no empty radio.
  - Optional field (`blank=True`): empty option remains.
  - Select widget behavior unchanged.
- Tests to add: tests/model_forms/test_modelchoicefield.py
  - test_modelform_radioselect_required_no_empty: cover unbound, bound-empty, and initial value scenarios using existing Article/Writer models.
- Edge cases: `null=True` still required at form level; admin RadioSelect subclasses also covered by `isinstance`.

We will implement per this spec and open a PR targeting `django__django-12406`.

### PR DESCRIPTION
**Pull request by casey-brooks**
Fixes #192

## Reproduction Steps
1. Ensure dependencies are installed (e.g. python -m venv, pip install asgiref sqlparse pytz).
2. From the repository root run:

   ```
   PYTHONPATH=$PWD DJANGO_SETTINGS_MODULE=tests.test_sqlite python tests/runtests.py model_forms.test_modelchoicefield.ModelChoiceFieldTests.test_modelform_radioselect_required_no_empty --parallel=1
   ```

## Observed Failure
```
Creating test database for alias 'default'...
F
======================================================================
FAIL: test_modelform_radioselect_required_no_empty (model_forms.test_modelchoicefield.ModelChoiceFieldTests.test_modelform_radioselect_required_no_empty)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/django/tests/model_forms/test_modelchoicefield.py", line 371, in test_modelform_radioselect_required_no_empty
    self.assertNotIn('value=""', rendered_required)
AssertionError: 'value=""' unexpectedly found in '<ul id="id_writer">\n    <li><label for="id_writer_0"><input type="radio" name="writer" value="" required id="id_writer_0" checked>\n ---------</label>\n\n</li>\n    <li><label for="id_writer_1"><input type="radio" name="writer" value="1" required id="id_writer_1">\n Writer 1</label>\n\n</li>\n    <li><label for="id_writer_2"><input type="radio" name="writer" value="2" required id="id_writer_2">\n Writer 2</label>\n\n</li>\n</ul>'

----------------------------------------------------------------------
Ran 1 test in 0.015s

FAILED (failures=1)
Destroying test database for alias 'default'...
Testing against Django installed in '/workspace/django/django'
System check identified no issues (0 silenced).
```

## Fix Summary
- suppress the synthetic empty radio option whenever a required ModelChoiceField is rendered with RadioSelect
- retain the empty option for optional RadioSelect fields to match blank=True behavior
- add regression coverage verifying required forms omit the empty radio and optional forms keep it without preselecting anything

## Testing
- `PYTHONPATH=$PWD DJANGO_SETTINGS_MODULE=tests.test_sqlite python tests/runtests.py model_forms.test_modelchoicefield.ModelChoiceFieldTests.test_modelform_radioselect_required_no_empty --parallel=1`
- `flake8 django/forms/models.py tests/model_forms/test_modelchoicefield.py`

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Test & Lint Summary
- `PYTHONPATH=$PWD DJANGO_SETTINGS_MODULE=tests.test_sqlite python tests/runtests.py model_forms.test_modelchoicefield.ModelChoiceFieldTests.test_modelform_radioselect_required_no_empty --parallel=1` *(1 passed, 0 failed, 0 skipped)*
- `flake8 django/forms/models.py tests/model_forms/test_modelchoicefield.py` *(no issues)*

File: tests/model_forms/test_modelchoicefield.py [hunk 1]
```diff
@@ -353,3 +353,30 @@ class CategoriesForm(forms.Form):
         )
         with self.assertNumQueries(2):
             template.render(Context({'form': CategoriesForm()}))
+
+    def test_modelform_radioselect_required_no_empty(self):
```
Comment (noa-lucent) [inline tests/model_forms/test_modelchoicefield.py:357]: [major] Thanks for adding coverage here. To make sure we meet the acceptance criteria from #192, we also need to exercise the bound-empty and initial-instance scenarios. Right now the test only inspects the fresh, unbound form (plus the optional-field case). Please extend the test to cover (1) a form bound to empty data, asserting it stays invalid and still renders without a `value=""` radio, and (2) a form bound to an existing `Article` instance where the correct writer radio is checked and no empty radio is rendered. That ensures the regression is locked down for all required behaviors.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
The field logic looks good, but the regression test needs to cover the additional scenarios called out in the issue spec.

File: tests/model_forms/test_modelchoicefield.py (hunk 1 above)
Comment (casey-brooks) [reply tests/model_forms/test_modelchoicefield.py:357]: Added bound-empty and instance-backed assertions to the regression test. The bound form stays invalid without rendering an empty or pre-checked radio, and the instance-backed form checks the saved writer with no empty option.

**Review by noa-lucent — APPROVED** [review]
Thanks for rounding out the regression coverage—the bound-empty and instance-backed cases look good now. The form logic continues to behave as expected for required and optional RadioSelects. No further issues from my side.
